test/blas1_test.hpp

```diff
@@ -0,0 +1,207 @@
+
+// the test class itself
+template <class B>
+class BLAS1_Test;
+
+template <class ScalarT_, class ExecutorType_>
+class BLAS1_Test<blas1_test_args<ScalarT_, ExecutorType_>>
+    : public ::testing::Test {
+ public:
+  using ScalarT = ScalarT_;
+  using ExecutorType = ExecutorType_;
+
+  BLAS1_Test() {}
+
+  virtual ~BLAS1_Test() {}
+  virtual void SetUp() {}
+  virtual void TearDown() {}
+
+  static size_t rand_size() {
+    // make sure the generated number is not too big for a type
+    // i.e. we do not want the sample size to be too big because of
+    // precision/memory restrictions
+    size_t ret = rand() >> 5;
+    int type_size =
+        sizeof(ScalarT) * CHAR_BIT - std::numeric_limits<ScalarT>::digits10 - 2;
+    return (ret & (std::numeric_limits<size_t>::max() +
+                   (size_t(1) << (type_size - 2)))) +
+           1;
+  }
+
+  // it is important that all tests are run with the same test size
+  // so each time we access this function within the same program, we get the
+  // same
+  // randomly generated size
+  size_t test_size() {
+    static bool first = true;
+    static size_t N;
+    if (first) {
+      first = false;
+      N = rand_size();
+    }
+    return N;
+  }
+
+  // getting the stride in the same way as the size above
+  size_t test_strd() {
+    static bool first = true;
+    static size_t N;
+    if (first) {
+      first = false;
+      N = ((rand() & 1) * (rand() % 5)) + 1;
+    }
+    return N;
+  }
+
+  template <typename DataType,
+            typename value_type = typename DataType::value_type>
+  static void set_rand(DataType &vec, size_t _N) {
+    value_type left(-1), right(1);
+    for (size_t i = 0; i < _N; ++i) {
+      vec[i] = value_type(rand() % int(right - left) * 1000) * .001 - right;
+    }
+  }
+
+  template <typename DataType,
+            typename value_type = typename DataType::value_type>
+  static void print_cont(const DataType &vec, size_t _N,
+                         std::string name = "vector") {
+    std::cout << name << ": ";
+    for (size_t i = 0; i < _N; ++i) std::cout << vec[i] << " ";
+    std::cout << std::endl;
+  }
+
+  template <typename DataType,
+            typename value_type = typename DataType::value_type>
+  static cl::sycl::buffer<value_type, 1> make_buffer(DataType &vec) {
+    return cl::sycl::buffer<value_type, 1>(vec.data(), vec.size());
+  }
+
+  template <typename value_type>
+  static vector_view<value_type, cl::sycl::buffer<value_type>> make_vview(
+      cl::sycl::buffer<value_type, 1> &buf) {
+    return vector_view<value_type, cl::sycl::buffer<value_type>>(buf);
+  }
+
+  template <typename DeviceSelector,
+            typename = typename std::enable_if<
+                std::is_same<ExecutorType, SYCL>::value>::type>
+  static cl::sycl::queue make_queue(DeviceSelector s) {
+    return cl::sycl::queue(s, [=](cl::sycl::exception_list eL) {
+      try {
+        for (auto &e : eL) {
+          std::rethrow_exception(e);
+        }
```

Comment:
note that only the first exception in the list will be processed, and the others will be discarded. Try catch should be inside the for loop to process all of them